qemu/tests/rebase_negative_test_on_iscsi.py

```diff
@@ -0,0 +1,120 @@
+import logging
+
+from virttest import qemu_storage
+from virttest import lvm
+from virttest import error_context
+from avocado.utils import process
+from autotest.client.shared import error
+
+
+class NegativeRebase(qemu_storage.QemuImg):
+
+    def __init__(self, test, params, env):
+        self.test = test
+        self.params = params
+        self.env = env
+        self.data_dir = "/dev/%s" % params.get("vg_name")
+        self.lvmdevice = None
+
+    def setup_lvm(self):
+        """
+        Setup a lvm environment and create logical volume:
+
+        :return logical volume object
+        """
+        self.params["pv_name"] = self.params.get("image_name")
+        lvmdevice = lvm.LVM(self.params)
+        error_context.context("Setup lvms", logging.info)
+        for lv_name in self.params.objects("lv_name_list"):
+            self.params["lv_name"] = lv_name
+            lvmdevice.setup()
+        self.lvmdevice = lvmdevice
+
+    def start_test(self):
+        """
+        Main function to run the negative test of rebase:
+        """
+        msg = ""
+        trigger_negative = False
+        rebase_chain = self.params.get("rebase_list", "").split(";")
+        self.setup_lvm()
+        error_context.context("Create images on lvms", logging.info)
+        for image_name in self.params.objects("images"):
+            filename = "%s/%s" % (self.data_dir,
+                                  self.params["image_name_%s" % image_name])
+            self.params.update(
+                {"image_name_%s" % image_name: filename,
+                 "image_size_%s" % image_name: self.params["lv_size"]})
+            t_params = self.params.object_params(image_name)
+            qemu_image = qemu_storage.QemuImg(t_params,
+                                              self.data_dir, image_name)
+            logging.info("Create image('%s') on %s." %
+                         (image_name, qemu_image.storage_type))
+            qemu_image.create(self.params)
+        error_context.context("Rebase snapshot to backingfile", logging.info)
+        for images in rebase_chain:
+            images = map(lambda x: x.strip(), images.split(">"))
+            try:
+                image = images[0]
+                base = images[1]
+            except IndexError:
+                msg = "Invalid format of'rebase_chain' params \n"
+                msg += "format like: 'image > base;image> base2;image2 > base2'"
+                break
+            self.params["image_chain"] = " ".join([base, image])
+            self.params["base_image_filename"] = image
+            try:
+                output = ""
+                t_params = self.params.object_params(image)
+                rebase_test = qemu_storage.QemuImg(t_params,
+                                                   self.data_dir, image)
+                cache_mode = self.params.get("cache_mode")
+                rebase_test.rebase(self.params, cache_mode)
+            except process.CmdError, err:
```

Comment:
As we have the way to make it work, I think It isn't worth us to modify avocado api for one negative case.